File: app/src/main/java/jp/gcreate/sample/kotlindaggerdatabinding/di/AppModule.java
package jp.gcreate.sample.kotlindaggerdatabinding.di;

import android.content.Context;

import javax.inject.Named;
import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;
import jp.gcreate.sample.kotlindaggerdatabinding.databases.OrmaDatabase;

/**
 * Copyright 2016 G-CREATE
 */
@Module
public class AppModule {
    private final Context context;

    public AppModule(Context context){
        this.context = context;
    }

    @Provides
    public Context provideContext(){
        return context;
    }

    @Provides
    @Singleton
    @Named("injectedString")
    public String provideInjectedString(){
        return "hoge from app module";
    }

    @Provides
    @Named("hogeString")
    public String provideHogeString(Context context){
        return context.getClass().getSimpleName();
    }

    @Provides
    @Singleton
    public OrmaDatabase provideOrmaDatabase(Context context){
        return OrmaDatabase.builder(context).build();
    }

}

File: app/src/main/java/jp/gcreate/sample/kotlindaggerdatabinding/di/AppModule.kt
package jp.gcreate.sample.kotlindaggerdatabinding.di

import android.content.Context
import android.databinding.DataBindingUtil
import android.view.LayoutInflater
import dagger.Module
import dagger.Provides
import jp.gcreate.sample.kotlindaggerdatabinding.R
import jp.gcreate.sample.kotlindaggerdatabinding.databases.TestData
import jp.gcreate.sample.kotlindaggerdatabinding.databinding.ActivityMainBinding
import javax.inject.Named
import javax.inject.Singleton

/**
 * Copyright 2016 G-CREATE

 */
@Module
class AppModule(private val context: Context) {

    @Provides
    fun provideContext(): Context {
        return context
    }

    @Provides
    @Singleton
    @Named("injectedString")
    fun provideInjectedString(): String {
        return "hoge from app module"
    }

    @Provides
    @Singleton
    @Named("contextString")
    fun provideContextString(context: Context): String{
        return context.javaClass.name
    }

    @Provides
    fun provideTestData(): TestData {
        return TestData(10, "from Dagger2")
    }

//    @Provides
//    @Singleton
//    fun provideOrmaDatabase(context: Context): OrmaDatabase {
//        return OrmaDatabase.builder(context).build()
//    }

    @Provides
    fun provideActivityMainBinding(context: Context): ActivityMainBinding{
        return DataBindingUtil.inflate(LayoutInflater.from(context), R.layout.activity_main, null, false)
    }
}